drivers/modem/quectel-bg9x.c

```diff
@@ -953,6 +953,64 @@ static const struct setup_cmd setup_cmds[] = {
 	SETUP_CMD_NOHANDLE("AT+QICSGP=1,1,\"" MDM_APN "\",\"" MDM_USERNAME "\", \"" MDM_PASSWORD "\",1"),
 };
 
+
+/* Func: modem_pdp_context_open
+ * Desc: This helper function is called from modem_setup, and is
+ * used to open the PDP context. If there is trouble opening the
+ *  PDP context, we try to deactive and reactive MDM_PDP_ACT_RETRY_COUNT times.
+ *  If it fails, we return an error.
+ */
+
+
+static int modem_pdp_context_open(const int contextID)
+{
+
+	int ret;
+	int pdp_act_retry_count = 0;
+	char buf_act[sizeof("AT+QIACT=##")] = {0};
+	char buf_deact[sizeof("AT+QIDEACT=##")] = {0};
+
+	if (contextID < MDM_MIN_PDP_CONTEXT_ID || contextID > MDM_MAX_PDP_CONTEXT_ID) {
+		LOG_ERR("WRONG CONTEXT ID USED : %d ; SHOULD BE BETWEEN %d AND %d",
+				contextID, MDM_MIN_PDP_CONTEXT_ID, MDM_MAX_PDP_CONTEXT_ID);
+				return -1;
+	}
+
+	snprintk(buf_act, sizeof(buf_act), "AT+QIACT=%d", contextID);
+	snprintk(buf_deact, sizeof(buf_deact), "AT+QIDEACT=%d", contextID);
```

Comment:
v2.5.0 is almost out and it would be great to include this patch. But to make that happen it would be better to limit changes to bugfix only and leave parameterization on `context_id` as a future enhancement. Also, we don't use any other context than 1, so there is no real value to add parameterization now, especially in a bugfix commit. This introduces new risk.

Reply:
Removed this part, left for future parameterization.